### PR TITLE
D2K - Add Ordos Mission 2a

### PR DESCRIPTION
**Pull request by MustaphaTR**
This is the first map that needs https://github.com/OpenRA/OpenRA/issues/12809. For now as i said at there, i'm using `OnEnteredFootprint`.

Our reinforcements are moved to a transition tile, so it won't be blocked by buildings we build. It can be moved to original location when https://github.com/OpenRA/OpenRA/issues/12464 is fixed.

### CODE REVIEW

**Comment by ltem** [comment]
...Review in progress...

I found some tile errors, I don't know if they are already fixed in #12784. I didn't see them because the map has in the gruntmod version a black padding and the OpenRA importer cannot cope with that.

I'll try to find an original map and test it.

 | OpenRA |
|:------:|:------:|
| image  | image |
| image  | image |
| image | image|
| image | image |

Code & Gameplay: I didn't found any issues till now.

**Comment by MustaphaTR** [comment]
I did the lazy way and just used existing similar tiles before that PR was out. I need to reimport the maps i guess.

**Comment by ltem** [comment]
Yep but this is also needed for the other multiplayer maps so, it would happen either way

File: mods/d2k/maps/ordos-02a/map.yaml [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+		LockColor: True
+		Color: B3EAA5
+		Enemies: Harkonnen, Creeps
+	PlayerReference@Harkonnen:
+		Name: Harkonnen
+		LockFaction: True
+		Faction: harkonnen
+		LockColor: True
+		Color: FE0000
+		Enemies: Ordos, Creeps
```
Comment (ltem) [inline mods/d2k/maps/ordos-02a/map.yaml]: Do we want to keep the `Creeps` as enemy? I hadn't any issues with the enemy hunting them but it has kind of proven itself in the other missions

**Review by ltem — COMMENTED** [review]
One small question, everything else looks fine.

File: mods/d2k/maps/ordos-02a/map.yaml (hunk 1 above)
Comment (MustaphaTR) [inline mods/d2k/maps/ordos-02a/map.yaml]: I thought i removed it already. I'll remove it when i find time.

**Comment by MustaphaTR** [comment]
Updated, but now depends on #12784.

**Comment by MustaphaTR** [comment]
Rebased.

**Review by ltem — APPROVED** [review]

**Comment by ltem** [comment]
:+1:

**Comment by reaperrr** [comment]
@MustaphaTR The Harkonnen deploys to HarkonnenRally6 can be blocked by placing a building there. If this is fixable through lua please do, otherwise because in my opinion carryalls should automatically move to the closest free cell in such a situation.
Otherwise this mission plays fine, so 👍 once that is fixed or filed as issue.

**Comment by MustaphaTR** [comment]
> The Harkonnen deploys to HarkonnenRally6 can be blocked by placing a building there. If this is fixable through lua please do, otherwise because in my opinion carryalls should automatically move to the closest free cell in such a situation.

#12464 

I can just change the location of the waypoint to an unbuildable cell for now tho.

**Comment by MustaphaTR** [comment]
Updated, moved that waypoint to a transition tile.

**Review by reaperrr — APPROVED** [review]

**Comment by reaperrr** [comment]
[changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/577031c35e9aeb9390775998146228be898533d7...5fdcd87cf433606b4b19c0224e42ab5f94947699)

**Pull request merged**